src/context/StoreContextProvider.tsx

```diff
@@ -1,17 +1,55 @@
+/* eslint-disable no-shadow */
 import { FC, ReactNode, useState } from 'react';
 import { StoreContext } from './StoreContext';
-import { PhoneCard } from '../types/PhoneCard';
+import { CartProductData } from '../types/CartProductData';
+import { CatalogProductData } from '../types/CatalogProductData';
+import { FullProductData } from '../types/FullProductData';
 
 type Props = {
   children: ReactNode;
 };
 
+const storagedCart = localStorage.getItem('cartContents');
+const savedCartContents = storagedCart ? JSON.parse(storagedCart) : [];
```

Comment:
this logic as I remember have to be in useEffect or in custom hook, I'm not sure but maybe it will be better to check